src/osd/PGBackend.cc

```diff
@@ -409,7 +409,8 @@ enum scrub_error_type PGBackend::be_compare_scrub_objects(
   if (auth.size != candidate.size) {
     if (error != CLEAN)
       errorstream << ", ";
-    error = SHALLOW_ERROR;
+    if (error != DEEP_ERROR)
+      error = SHALLOW_ERROR;
```

Comment:
@liewegas In addition to your change at line 412 we should not set read_error in be_scan_list() on stat() errors and set the stat size to something invalid instead.   Also, remove the comment at line 374 and change read_error to a DEEP_ERROR.

This way a regular scrub will never get a DEEP_ERROR from this function, but a deep-scrub could possibly see a SHALLOW_ERROR if only stat() fails.  This keeps the error counts from getting messed up.